### PR TITLE
Add `tokenizer-only` flag to only download tokenizers from HF or oras

### PR DESCRIPTION
**Pull request by irenedea**
## Manual tests
Download from oras: `test-dl-tokenizer-m3T8eq`
Download from hf: `test-dl-tokenizer-inG1Zp`

### CODE REVIEW

File: scripts/misc/download_model.py
Comment (jerrychen109) [inline scripts/misc/download_model.py:14]:
```suggestion
    python download_model.py http --url https://server.com/models/mosaicml/mpt-7b/ --save-dir <save_dir>
```

File: scripts/misc/download_model.py
Comment (jerrychen109) [inline scripts/misc/download_model.py:62]:
```suggestion
    base_parser.add_argument('--tokenizer-only',
                             type=bool,
                             default=False,
                             action='store_true')
```

File: scripts/misc/download_model.py
Comment (jerrychen109) [inline scripts/misc/download_model.py:92]:
```suggestion
        if args.tokenizer_only:
```
If we specify the type and have the default, it'll be either true or false

File: llmfoundry/utils/model_download_utils.py
Comment (jerrychen109) [inline llmfoundry/utils/model_download_utils.py:235]:
```suggestion
                       tokenizer_only: bool = False,
```
for consistency with `download_from_hf_hub`

File: llmfoundry/utils/model_download_utils.py
Comment (jerrychen109) [inline llmfoundry/utils/model_download_utils.py:67]:
```suggestion
        tokenizer_only (bool): If true, only download tokenizer files.
```
Thinking singular makes more sense?

File: llmfoundry/utils/model_download_utils.py
Comment (jerrychen109) [inline llmfoundry/utils/model_download_utils.py:106]:
```suggestion
    allow_patterns = TOKENIZER_FILES if tokenizer_only else None
```
For consistency with above suggestion

File: scripts/misc/download_model.py
Comment (jerrychen109) [inline scripts/misc/download_model.py:119]:
```suggestion
                             tokenizer_only=args.tokenizer_only,
```
For consistency with singular `tokenizer_only`

File: scripts/misc/download_model.py
Comment (jerrychen109) [inline scripts/misc/download_model.py:123]:
```suggestion
                           args.save_dir, tokenizers_only=args.tokenizer_only, args.concurrency)
```
more consistency

**Review by jerrychen109 — COMMENTED** [review]
mostly lgtm - added some suggestions for consistency in the code. Let me know if they make sense and I'll approve after!

File: scripts/misc/download_model.py
Comment (irenedea) [reply scripts/misc/download_model.py:62]: apparently, you cannot specify type with action. I initially had [that](https://github.com/mosaicml/llm-foundry/pull/895/commits/936d5e566d0db411f7f5a93e3e2bffce4194b109#diff-b91dc3e13956a84272e7cb09c99a864c10739a26142341d239ffd53596a887eeR59), but had to remove.

File: scripts/misc/download_model.py
Comment (irenedea) [reply scripts/misc/download_model.py:92]: yeah with the above comment, I cannot specify the type with store_action. Maybe that's safe enough? I just preferred to be explicit in case. lmk what you think

File: llmfoundry/utils/model_download_utils.py
Comment (irenedea) [reply llmfoundry/utils/model_download_utils.py:67]: ah yes you're right! good catch

File: scripts/misc/download_model.py
Comment (jerrychen109) [reply scripts/misc/download_model.py:92]: I think it's still safe with the default value since `store_true` implies bool already so I don't think a user can provide a weird value, but not super opinionated here. I'll leave it to you!

**Review by jerrychen109 — APPROVED** [review]

**Pull request merged**
